src/core_landice/mode_forward/Interface_velocity_solver.cpp

```diff
@@ -1157,7 +1160,7 @@ void importFields(std::vector<std::pair<int, int> >& marineBdyExtensionMap,  dou
       int v = verticesOnTria[iVertex + 3 * index];
       int iCell = vertexToFCell[v];
       //compute temperature by averaging temperature values of triangles vertices where ice is present
-      if (cellsMask_F[iCell] & ice_present_bit_value) {
+      if (thickness_F[iCell] > thermal_thickness_limit * unit_length) {
```

Comment:
You do not have to scale this by `unit_length`. `thickness_F` is the original MPAS thickness.

Reply:
I did this because when I set the value of thermal_thickness_limit I decided to have it use Albany units (km) for consistency with the other things we import from MPAS.  So here I'm converting it back to meters because the thickness is in m.

Reply:
Ah, OK!  Maybe it's worth putting a commet to explaining that.